contracts/script/SetupPayments.s.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import {Script} from "forge-std/Script.sol";
+import {HelloWorldDeploymentLib} from "./utils/HelloWorldDeploymentLib.sol";
+import {CoreDeploymentLib} from "./utils/CoreDeploymentLib.sol";
+import {Merkle} from "@eigenlayer/contracts/libraries/Merkle.sol";
+import {IRewardsCoordinator} from "@eigenlayer/contracts/interfaces/IRewardsCoordinator.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IStrategy} from "eigenlayer-contracts/src/contracts/interfaces/IStrategyManager.sol";
+
+
+contract SetupPayments is Script {
+
+    address private deployer;
+    address proxyAdmin;
+    CoreDeploymentLib.DeploymentData coreDeployment;
+    HelloWorldDeploymentLib.DeploymentData helloWorldDeployment;
+
+    uint256 NUM_PAYMENTS = 8;
+    uint256 NUM_TOKEN_EARNINGS = 1;
+    uint256 TOKEN_EARNINGS = 100;
+    
+    struct PaymentLeaves{
+        bytes32[] leaves;
+        bytes32[] tokenLeaves;
+    }
+
+    function setUp() public {
+        deployer = vm.rememberKey(vm.envUint("PRIVATE_KEY"));
+        vm.label(deployer, "Deployer");
+
+        coreDeployment = CoreDeploymentLib.readDeploymentJson("deployments/core/", block.chainid);
+        helloWorldDeployment = HelloWorldDeploymentLib.readDeploymentJson("deployments/hello-world/", block.chainid);
+    }
+
+    function run() external {
+        vm.startBroadcast(deployer);
+        //set rewardsUpdater to be the deployer
+        IRewardsCoordinator(coreDeployment.rewardsCoordinator).setRewardsUpdater(deployer);
+
+        vm.stopBroadcast();
+    }
+    //submits the payments on behalf of the AVS
+    function createPaymentSubmissions(uint256 numPayments, uint256 amountPerPayment, uint32 duration) public {
```

Comment:
overall structure lgtm

Can you refactor most of this logic into a library and then use the library to run a test for rewardSubmission and also claiming the payment

This forge script itself should be pretty minimal logic-wise

Reply:
Added separate "lib" file